jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -191,4 +204,50 @@ private static FileEntriesLayer addDirectoryContentsToLayer(
             });
     return builder.build();
   }
+
+  private static List<FileEntriesLayer> getDependenciesLayers(Path jarPath) throws IOException {
+    List<FileEntriesLayer> layers = new ArrayList<>();
+    String classPath = null;
+
+    // Get dependencies from Class-Path in the jar's manifest and add a layer each for non-snapshot
+    // and snapshot dependencies. If Class-Path is not present in the jar's manifest then skip
+    // adding the dependencies layers.
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      classPath = jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.CLASS_PATH);
+    }
+    if (classPath != null) {
+      Predicate<String> isSnapshot = name -> name.contains("SNAPSHOT");
+      List<String> allDependencies = Splitter.onPattern("\\s+").splitToList(classPath.trim());
+      List<Path> nonSnapshotDependencies =
+          allDependencies
+              .stream()
+              .filter(isSnapshot.negate())
+              .map(Paths::get)
+              .collect(Collectors.toList());
+      List<Path> snapshotDependencies =
+          allDependencies.stream().filter(isSnapshot).map(Paths::get).collect(Collectors.toList());
+      Path jarParent = jarPath.getParent();
+      if (!nonSnapshotDependencies.isEmpty()) {
+        FileEntriesLayer.Builder nonSnapshotDependenciesLayerBuilder =
+            FileEntriesLayer.builder().setName(DEPENDENCIES);
+        nonSnapshotDependencies.forEach(
+            path ->
+                nonSnapshotDependenciesLayerBuilder.addEntry(
+                    jarParent.resolve(path),
```

Comment:
I just tried my `C.jar` example, but it throws an NPE. In this case, `jarPath` is simply `C.jar` (not an absolute path) and `jarPath.getParent()` returns null.

```
~/A$ jib --target=docker://foo --stacktrace jar C.jar
java.lang.NullPointerException
        at com.google.cloud.tools.jib.cli.jar.JarModeProcessor.lambda$getDependenciesLayers$6(JarModeProcessor.java:236)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at com.google.cloud.tools.jib.cli.jar.JarModeProcessor.getDependenciesLayers(JarModeProcessor.java:233)
```

Maybe it can be
```java
      Path jarParent = jarPath.getParent() == null ? Paths.get("") : jarPath.getParent();
```

Reply:
I ran into the same issue, you're right. Okay I will fix this and add some tests for jars that use dependencies.